### PR TITLE
Test for login page component

### PR DESCRIPTION
**Pull request by steve-237**
Still in progress. Need help finding useful test cases.

### CODE REVIEW

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts [hunk 1]
```diff
@@ -57,4 +72,42 @@ describe('LoginPageComponent', () => {
     let test_div = de.query(By.css('.test-card'));
     expect(test_div).toBeNull();
   });
+
+  it('should have an initial state like', () => {
```
Comment (Idontker) [inline src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:76]: Description does not describe a test/behavior

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts [hunk 2]
```diff
@@ -57,4 +72,42 @@ describe('LoginPageComponent', () => {
...
+
+  it('empty input form is invalid', (() => {
+    let form = component.formGroup;
+    expect(fixture.debugElement.nativeElement.querySelector('input').innerHTML).toBe("");
+    expect(form.invalid).toBeTruthy();
+  }));
+
+  it('should create a form with email, password field and a button', () => {
+    let email = fixture.debugElement.nativeElement.querySelector('input');
+    let password = fixture.debugElement.nativeElement.querySelector('input');
```
Comment (Idontker) [inline src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:89]: password and email query the same Element

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts [hunk 3]
```diff
@@ -57,4 +72,42 @@ describe('LoginPageComponent', () => {
...
   });
+
+  it('should have an initial state like', () => {
+    expect(de.query(By.css('input'))).toBeDefined();
+    expect(component.formGroup.invalid).toBeTruthy();
+  });
+
+  it('empty input form is invalid', (() => {
+    let form = component.formGroup;
+    expect(fixture.debugElement.nativeElement.querySelector('input').innerHTML).toBe("");
```
Comment (Idontker) [inline src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:83]: there are multiple input fields within the HTML. This does not test the intended behavior.

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts [hunk 4]
```diff
@@ -57,4 +72,42 @@ describe('LoginPageComponent', () => {
...
+    spyLoginPostRequest.calls.reset();
+    expect(spyLoginPostRequest).not.toHaveBeenCalled();
+
+    component.formGroup.controls['email'].setValue('test@fau.de');
+    component.formGroup.controls['password'].setValue('test');
+
+    component.loginProcess();
+
+    expect(component.formGroup.valid).toBeTruthy();
+    expect(spyLoginPostRequest).toHaveBeenCalled();
```
Comment (Idontker) [inline src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:107]: Missing checks/expectation for parameters of the loginPostRequest call

**Review by Idontker — CHANGES_REQUESTED** [review]
Missing tests for the following behaviors: 

- loginPostRequest redirects on successful login
- loginPostRequest opens PupUpComponent on failed login request.
- loginPostRequest is not called when formGroup is invalid

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts (hunk 1 above)
Comment (steve-237) [reply src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:76]: Does the test even make sense? If so, what would be an appropriate description?

Comment (Idontker) [reply src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:76]: How can I tell whether your intended test is usefull? 
Nethertheless, this mini tests does not appear usefull

**Comment by Idontker** [comment]
@steve-237 are you still working on this task?

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts (hunk 2 above)
Comment (steve-237) [reply src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts:89]: Mistake. I forgot to specify the right element. Thanks for your points! I will work on it again

**Pull request closed without merging**